metric/system/process/process.go

```diff
@@ -191,23 +191,28 @@ func (procStats *Stats) GetSelf() (ProcState, error) {
 
 // pidIter wraps a few lines of generic code that all OS-specific FetchPids() functions must call.
 // this also handles the process of adding to the maps/lists in order to limit the code duplication in all the OS implementations
-func (procStats *Stats) pidIter(pid int, procMap ProcsMap, proclist []ProcState) (ProcsMap, []ProcState) {
+func (procStats *Stats) pidIter(pid int, procMap ProcsMap, proclist []ProcState) (ProcsMap, []ProcState, error) {
 	status, saved, err := procStats.pidFill(pid, true)
+	var nonFatalErr error
 	if err != nil {
 		if !errors.Is(err, NonFatalErr{}) {
 			procStats.logger.Debugf("Error fetching PID info for %d, skipping: %s", pid, err)
-			return procMap, proclist
+			if canIgnore(err) {
```

Comment:
```suggestion
	        // While monitoring a set of processes, some processes might get killed after we get all the PIDs
	        // So, there's no need to capture "process not found" error.
			if errors.Is(err, syscall.ESRCH) {
```

This function only gets called one time, the implementation is a single line, and the comment provides useful context that shouldn't be hidden behind a function call. I'd just use this directly in the single place it gets referenced.

Reply:
Sure. That makes sense!